src/form.rs

```diff
@@ -62,8 +62,8 @@ pub enum MetaForm {}
 
 impl Form for MetaForm {
 	type String = &'static str;
-	type TypeId = MetaType;
-	type IndirectTypeId = MetaType;
+	type Type = MetaType;
+	type IndirectType = MetaType;
```

Comment:
So what we still need here as `Form` associated types is `String` (unchanged) and `TypeId` (either `core::any::TypeId` or the compressed `TypeId` you get from the interners). Note that `Type` is not completely transparent here because this is mainly about how we communicate about type connections (e.g. via their IDs), not how the types are layed out which is described by the crate's `Type` structs and variants. (formerly `TypeDef` and `TypeId` that are now merged)